backend/api/llm_providers.py

```
@@ -45,17 +45,15 @@ def _get_fingerprint(api_key: str | None) -> str | None:
-    user_roles = ["admin"] if user_id == "admin" else []
-    if "admin" not in user_roles:
-        raise HTTPException(status_code=403, detail="Admin access required")
-    return user_id
+async def check_admin_access(auth_context: AuthContext = Depends(get_auth_context)):
+    if auth_context.role not in {"platform_admin", "organization_admin"}:
+        raise HTTPException(status_code=403, detail="Organization admin access required")
+    return auth_context.user_id
 
 @router.get("", response_model=List[LLMProviderResponse])
 async def list_providers(
     db: AsyncSession = Depends(get_db),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)